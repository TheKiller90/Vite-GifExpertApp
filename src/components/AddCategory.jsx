import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target:{ value } }) => {
        setInputValue(value);
    }
    const onSubmit = e => {
        e.preventDefault();
        
        const cleanInputValue = inputValue.trim();
        if(cleanInputValue.length <= 1) return;
        
        onNewCategory(cleanInputValue);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
                autoFocus
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};