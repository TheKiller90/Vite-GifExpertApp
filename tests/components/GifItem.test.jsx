import { render } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';

describe('Evaluando el componente gifItem', () => {
    const title = 'Mono';
    const url = 'https://mono.io';

    test('Hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect(container).toMatchSnapshot();
    });
});