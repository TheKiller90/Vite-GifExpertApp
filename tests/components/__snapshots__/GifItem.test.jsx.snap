// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluando el componente gifItem Hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mono"
      src="https://mono.io"
    />
    <p>
      Mono
    </p>
  </div>
</div>
`;
